### DISCUSSION TITLE
How to make independently detect collision when using animation sprite?

### DISCUSSION THREAD

**Discussion by doowonee** [flame-engine/flame#1912] [Q&A] [answered]
I add something like claw animation sprite below. It is a melee attack with 3 frames.

https://user-images.githubusercontent.com/10105141/190859264-fb703b95-a3d9-47d4-83c2-cb7bb439c848.mp4

How should I add hitbox to detect collision exactly same area with animation?

![w1](https://user-images.githubusercontent.com/10105141/190859336-ffa0ffa0-f2e9-4f3c-bcec-caae3dc23db9.png)

Maybe should I make parts of weapon separately above as a components contain hitbox and make my own animation via changing `transform` in `update` method?

**Comment by spydon** [comment, marked as answer]
Just like you say, I would create a separate component with a hitbox for the weapon and then add a `MoveEffect` on that one. Or you have to change the positioning of the hitbox manually when the animation is running.

**Reply by doowonee** [reply]
what a fast answer!

What about rotate? Is it possible to using multiple effects in same time?

I should change position and also angle to implement the animation

**Reply by ufrshubham** [reply]
> Is it possible to using multiple effects in same time?

Yes, it work just fine.


> I should change position and also angle to implement the animation

Depends on what kinds of animation you are trying to achieve, but it sounds about right in your case. I recently used something similar for a character that can attack using a sword. I used a separate component for the sword and attached it as child of the main character. I also added a hitbox for the sword and while attacking I just applied a rotate effect to the sword. It automatically moves the hitbox along with the sword.

**Reply by doowonee** [reply]
thanks for your insight!
